File: app/src/main/java/com/valhalla/lolchampviewer/net/models/Champion.kt
package com.valhalla.lolchampviewer.net.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Champion(
    @SerialName("id") val id: String,
    @SerialName("key") val key: String,
    @SerialName("name") val name: String,
    @SerialName("title") val title: String,
    @SerialName("image") val image: Image,
    @SerialName("skins") val skins: List<Skin>,
    @SerialName("lore") val lore: String,
    @SerialName("blurb") val blurb: String,
    @SerialName("allytips") val allTips: List<String>,
    @SerialName("enemytips") val enemyTips: List<String>,
    @SerialName("tags") val tags: List<String>,
    @SerialName("partype") val partype: String,
    @SerialName("info") val info: Info,
    @SerialName("stats") val stats: Stats
): java.io.Serializable

File: egg/lib/models/Champion.dart
import 'package:json_annotation/json_annotation.dart';

import 'Skin.dart';
import 'Image.dart';
import 'Info.dart';
import 'Stats.dart';
import 'Spell.dart';
import 'Passive.dart';

part 'Champion.g.dart';

@JsonSerializable()
class Champion {
  @JsonKey(name: "id") final String id ;
  @JsonKey(name: "key") final String key ;
  @JsonKey(name: "name") final String name ;
  @JsonKey(name: "title") final String title ;
  @JsonKey(name: "image") final Image image ;
  @JsonKey(name: "skins") final List<Skin> skins ;
  @JsonKey(name: "lore") final String lore ;
  @JsonKey(name: "blurb") final String blurb ;
  @JsonKey(name: "allytips") final List<String> allTips ;
  @JsonKey(name: "enemytips") final List<String> enemyTips ;
  @JsonKey(name: "tags") final List<String> tags ;
  @JsonKey(name: "partype") final String partype ;
  @JsonKey(name: "info") final Info info ;
  @JsonKey(name: "stats") final Stats stats ;
  @JsonKey(name: "spells") final List<Spell> spells ;
  @JsonKey(name: "passive") final Passive passive ;

  Champion({
    this.id,
    this.key,
    this.name,
    this.title,
    this.image,
    this.skins,
    this.lore,
    this.blurb,
    this.allTips,
    this.enemyTips,
    this.tags,
    this.partype,
    this.info,
    this.stats,
    this.spells,
    this.passive,
  });

  factory Champion.fromJson(Map<String, dynamic> json) => _$ChampionFromJson(json);
  Map<String, dynamic> toJson() => _$ChampionToJson(this);
}
